packages/@textlint/textlint-plugin-text/src/TextProcessor.ts

```diff
@@ -0,0 +1,35 @@
+/* eslint-disable no-unused-vars */
+// LICENSE : MIT
+"use strict";
+import { parse } from "@textlint/text-to-ast";
+import { TextlintPluginProcessor, TextlintPluginOptions } from "@textlint/types";
+import { TxtParentNode } from "@textlint/ast-node-types/src";
+
+export class TextProcessor implements TextlintPluginProcessor {
+    config: TextlintPluginOptions;
+    extensions: Array<string>;
+    constructor(config = {}) {
+        this.config = config;
+        // support "extension" option
+        this.extensions = this.config.extensions ? this.config.extensions : [];
+    }
+
+    availableExtensions() {
+        return [".txt", ".text"].concat(this.extensions);
+    }
+
+    processor(_ext: string) {
+        return {
+            preProcess(text: string, _filePath?: string): TxtParentNode | { text: string; ast: TxtParentNode } {
+                // FIXME: remove any
+                return parse(text) as any;
```

Comment:
https://github.com/textlint/textlint/blob/953eb036d4a4b6a3249a2eaaa9af27e4da7b69ea/packages/%40textlint/types/src/Plugin/TextlintPluginModule.ts#L45
TextlintPluginProcessor.preProcess need return `TxtParentNode | { text: string; ast: TxtParentNode }` but `parse()` from "@textlint/ast-node-types" return `TxtNode` type. 

So I couldn't resolve this mismatch of types. Could you give me some advice on how to resolve it?

Reply:
Actually, https://github.com/textlint/textlint/blob/953eb036d4a4b6a3249a2eaaa9af27e4da7b69ea/packages/%40textlint/text-to-ast/src/plaintext-parser.ts#L128 `parse` return a `TxtParentNode`.
It means that the returned AST has `children` property.

Reply:
Probably, The truth is that `preProcess` return type is wrong.
textlint's core also support `TxtNode`(not have `children` property) as `preProcess`'s return value. possibly.

We have two options.

1. fix `preProcess` return type
2. fix parse return type

Reply:
Unfortunately, I don't know the entire of textlint implementation. So I'll follow your decision.

Reply:
Can you fix `preProcess` return type?

```diff
diff --git a/packages/@textlint/types/src/Plugin/TextlintPluginModule.ts b/packages/@textlint/types/src/Plugin/TextlintPluginModule.ts
index 923dc1c9..436ed944 100644
--- a/packages/@textlint/types/src/Plugin/TextlintPluginModule.ts
+++ b/packages/@textlint/types/src/Plugin/TextlintPluginModule.ts
@@ -1,5 +1,5 @@
 // Plugin
-import { TxtParentNode } from "@textlint/ast-node-types";
+import { TxtNode } from "@textlint/ast-node-types";
 
 /**
  * textlint plugin option values is object or boolean.
@@ -42,7 +42,7 @@ export declare class TextlintPluginProcessor {
          * @param text
          * @param filePath
          */
-        preProcess(text: string, filePath?: string): TxtParentNode | { text: string; ast: TxtParentNode };
+        preProcess(text: string, filePath?: string): TxtNode | { text: string; ast: TxtNode };
         postProcess(messages: Array<any>, filePath?: string): { messages: Array<any>; filePath: string };
     };
 }

```

Reply:
I fixed at 7880037
Is this BREAKING CHANGES? I just filled "chore" in the commit message, but if it's wrong, fix it.

Reply:
It is not breaking change.
If A node which is suitable for TxtParentNode, the node is suitable for TxtNode.

Also typing change is minor in textlint
https://github.com/textlint/textlint#semantic-versioning-policy

Reply:
All right. Should I fix this commit message to "feat(types)" following conventional commit rule and push force?

Reply:
No problem.